lib/shared/addon/components/form-members/component.js

```diff
@@ -143,20 +142,44 @@ export default Component.extend({
       }
 
     });
-    // console.log(add, update, remove);
 
-    return PromiseAll(add.map(x => x.save())).then(() => {
-      return PromiseAll(update.map(x => x.save())).then(() => {
-        return PromiseAll(remove.map(x => x.delete())).then(() => {
-          if ( this.isDestroyed || this.isDestroying ) {
-            return;
-          }
+    if (this.hasOwner(remove, add)) {
+      return PromiseAll(add.map(x => x.save())).then(() => {
+        return PromiseAll(update.map(x => x.save())).then(() => {
+          return PromiseAll(remove.map(x => x.delete())).then(() => {
+            if ( this.isDestroyed || this.isDestroying ) {
+              return;
+            }
 
-          set(this, 'memberArray', []);
-          return get(this, 'primaryResource');
+            set(this, 'memberArray', []);
+            return get(this, 'primaryResource');
+          });
         });
       });
-    });
+    } else {
+      this.set('errors', [get(this, 'intl').t('formMembers.members.errors.ownerReq')]);
+      return reject();
+    }
+  },
+
+  hasOwner: function(toRemove, toAdd) {
```

Comment:
This doesn't sound like it works right..

- Only rows being removed or added are considered.  If you have 5 owners and remove 1, there's no row in toAdd so it always returns false (or true, if toAdd is somehow null...)

- You could be in a situation where there is currently no owner (e.g. because the only owner quit and was deleted).  This doesn't check that there is _actually_ an owner in the list, it only attempts to check that _if_ you are removing an owner, there is still another.

Can't it just loop over the currently displayed list of bindings and ensure that one of them is owner?